R/utils.R

```diff
@@ -17,24 +17,47 @@ formatSolveBioResponse <- function (res, raw = FALSE) {
         res$'_url' = url
         return(res)
     }
+
 }
 
-formatSolveBioQueryResponse <- function (res, raw = FALSE, row.names = NULL) {
+formatSolveBioQueryResponse <- function (id, res, raw = FALSE, row.names = NULL) {
     # res will be the output of formatSolveBioResponse
     if (!raw & is.data.frame(res$results)) {
-        # Flatten the data frame
+        # Append column "_id" if it's not already there
+        if (!"_id" %in% row.names) {
```

Comment:
We should make sure col.names (formerly row.names) is not NULL

Reply:
Done.